pkgs/applications/virtualization/nvidia-docker/libnvc.nix

```diff
@@ -0,0 +1,55 @@
+{ stdenv, lib, fetchFromGitHub, fetchurl, libelf, libcap, libseccomp
+, linuxPackages }:
+
+with lib; let
+
+  persistenced = linuxPackages.nvidia_x11.passthru.persistenced;
+
+  nvidia-modprobe = fetchurl {
+    url = "https://github.com/NVIDIA/nvidia-modprobe/archive/396.51.tar.gz";
+    sha256 = "082b1jvkg1dij6l7yvmi7vi1fdbmp7jw4snpkw76ggl4lcvn9g15";
+  };
+
+in stdenv.mkDerivation rec {
+  name = "libnvidia-container-${version}";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = "libnvidia-container";
+    rev = "v1.0.0";
+    sha256 = "1ws6mfsbgxhzlb5w1r8qqg2arvxkr21n59i4cqsyz3h5jsqsflbw";
+  };
+
+  # locations of nvidia-driver libraries are not resolved via ldconfig which
+  # doesn't get used on NixOS. Additional support binaries like nvidia-smi are
+  # not resolved via the environment PATH but via the derivation output path.
+  patches = [ ./libnvc-ldconfig-and-path-fixes.patch ];
+
+  makeFlags = [
+    "WITH_LIBELF=yes"
+    "prefix=$(out)"
+  ];
+
+  postPatch = ''
+    sed -i 's/^REVISION :=.*/REVISION = ${src.rev}/' mk/common.mk
+    sed -i 's/^COMPILER :=.*/COMPILER = $(CC)/' mk/common.mk
+
+    mkdir -p deps/src/nvidia-modprobe-396.51
+    tar -C deps/src/nvidia-modprobe-396.51 --strip-components=1 -xz \
+      -f  ${nvidia-modprobe} nvidia-modprobe-396.51/modprobe-utils
+    touch deps/src/nvidia-modprobe-396.51/.download_stamp
+
+    # Replace searching PATH for nvidia-smi etc. with output/mnt copies
+    substituteInPlace src/nvc_info.c --replace "@out@" $out
+  '';
+
+  buildInputs = [ libelf libcap libseccomp linuxPackages.nvidia_x11 ];
+
+  preInstall = ''
+    install -Dm755 -t $out/mnt ${linuxPackages.nvidia_x11.bin}/bin/* ${persistenced}/bin/*
```

Comment:
Why use $out/mnt for binaries?

Reply:
Those are binaries that are mounted into the container (and therefore patched to work inside), but that probably doesn't make sense for naming the source, so it's just /bin now.